### PR TITLE
Bringing helm demo up to date with latest version (>= v29) of Jenkins helm chart

### PR DESCRIPTION
**Pull request by holmesb**
Signed-off-by: Brendan Holmes <brendan.holmes@asos.com>

Here is our checklist for contributors. No hard requirement here, just a reminder

- [x] Please describe what you did
Updated helm demo in-line with latest Helm chart

- [x] Link to issue you're working on if there's a relevant one
[#675 ](https://github.com/jenkinsci/configuration-as-code-plugin/issues/675)

- [x] Did you provide a test-case to demonstrate feature actually works / issue is fixed ?
Yes checked Jenkins CasC with auto-reload works with these values.

Many of the non-JCasC values in the Jenkins helm chart with shortly be [deprecated ](https://github.com/helm/charts/issues/11276)in favour of using JCasC instead.  When this is done I'll submit another PR to update this demo accordingly.  But this new documentation & values.yaml are accurate for the time being.

### CODE REVIEW

File: demos/kubernetes-helm/README.md [hunk 1]
```diff
@@ -20,6 +20,53 @@ Master:
...
+        #     memory: 50Mi
+      # SSH port value can be set to any unused TCP port.  The default, 1044, is a non-standard SSH port that has been chosen at random.
+      # Is only used to reload jcasc config from the sidecar container running in the Jenkins master pod.
+      # This TCP port will not be open in the pod (unless you specifically configure this), so Jenkins will not be
+      # accessible via SSH from outside of the pod.  Note if you use non-root pod privileges (RunAsUser & FsGroup),
+      # this must be > 1024:
+      sshTcpPort: 1044
+      # label that the configmaps with dashboards are marked with:
+      label: jenkins_config
+      # folder in the pod that should hold the collected dashboards:
```
Comment (timja) [inline demos/kubernetes-helm/README.md]: what do you mean by dashboards?

Comment (holmesb) [reply demos/kubernetes-helm/README.md]: Good spot.  I have corrected.  Some of the helm code was from the Grafana chart (has dashboards).

**Comment by timja** [comment]
Your git user config doesn't look right @holmesb 
Its not linking to your github account
https://help.github.com/articles/setting-your-username-in-git/

**Comment by holmesb** [comment]
I've set my git username to holmesb in this repo now.  Hopefully resolves.

**Comment by timja** [comment]
it'll need a `git commit --amend --reset-author` and a force push

**Comment by casz** [comment]
@holmesb 
you might want to look into fixing your git user.email config 😅 
or set up an email alias on your GitHub account 🥇 
to get proper credit for your commits 🏆 
https://help.github.com/articles/adding-an-email-address-to-your-github-account/
https://help.github.com/articles/setting-your-commit-email-address-in-git
then to amend your commits: https://stackoverflow.com/a/3042512

**Comment by casz** [comment]
@ewelinawilkosz 😢 he never actually managed to fix his commits as noticed by the missing profile picture on the commits.

**Comment by casz** [comment]
@holmesb you properly just have to associate your email with your GitHub account:

**Comment by holmesb** [comment]
@ewelinawilkosz @casz Ah ok, done now.  Should be fine from now on.  Thx

**Comment by casz** [comment]
@holmesb thanks, and thank you for your contribution 🙌

**Pull request merged**